routes/linkUrls.js

```diff
@@ -0,0 +1,33 @@
+const express = require('express');
+const router = express.Router();
+const cache = require('memory-cache');
+const asyncHandler = require('express-async-handler');
+
+const handleCache = require('../helpers/handleCache');
+const commonContent = require('../helpers/commonContent');
+
+router.get('/:codenames', asyncHandler(async (req, res, next) => {
+  await handleCache.evaluateCommon(res, ['urlMap']);
+
+  const KCDetails = commonContent.getKCDetails(res);
+  const urlMap = await cache.get(`urlMap_${KCDetails.projectid}`);
+  const codenames = req.params.codenames.split('/');
+
+  if (codenames.length > 0) {
+    let resolvedUrl = [];
+
+    if (urlMap) {
+      resolvedUrl = urlMap.filter(elem => elem.codename === codenames[0]);
+    }
+
+    if (resolvedUrl.length > 0) {
+      const url = resolvedUrl[0].url;
+
+      return res.redirect(303, url);
+    } else {
+      return next();
```

Comment:
What's the next middleware? Or where will this end? (In 404 or somewhere else?)

Reply:
Discussed in person - both returning "nothing" and calling `next()` makes the web end up in 404, which is according to the desired behavior. I will unify the approach though, and this endpoint will always end up in calling `next()` in case of an invalid path.